### ISSUE TITLE
Bug: 'object-shorthand' rule breaks when the arguments object is used

### ISSUE THREAD

**Issue by thoughtsunificator** [eslint/eslint#19191]
### Environment

Node version: v23.1.0
npm version: 10.9.0
Local ESLint version: 9.13.0
Operating System:  Linux x64

### What parser are you using?

Default (Espree)

### What did you do?

The following code do not pass the checks for the rule 'object-shorthand' when running eslint 9.13.0:

```js
function foo() {
 const bar = []
 bar.push({ arguments })
}
```

### What did you expect to happen?

The code should pass the rule check.

### What actually happened?

Unexpected mix of shorthand and non-shorthand properties

### Link to Minimal Reproducible Example

see above

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Please provide a minimal reproducible example with your configuration for this rule. You can use the [ESLint Playground](https://eslint.org/play/) and paste a link to it.
